src/Microsoft.SqlTools.ServiceLayer/ShowPlan/ShowPlanGraph/XmlPlanNodeBuilder.cs

```diff
@@ -371,6 +374,99 @@ private IEnumerable<BaseStmtInfoType> EnumStatements(StmtBlockType statementBloc
             }
         }
 
+        private Description ParseDescription(ShowPlanGraph graph, int index)
+        {
+            XmlDocument stmtXmlDocument = new XmlDocument();
```

Comment:
I would at least indicate where you got this logic from - and it would be great if you could spend some time understanding what it's doing and comment it appropriately. It's a bit dense and hard to understand at a glance.